internal/ctrlflow/hardening.go

```diff
@@ -0,0 +1,287 @@
+package ctrlflow
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	mathrand "math/rand"
+	"strconv"
+
+	"golang.org/x/exp/rand"
+	"golang.org/x/tools/go/ssa"
+	ah "mvdan.cc/garble/internal/asthelper"
+	"mvdan.cc/garble/internal/literals"
+)
+
+var hardeningMap = map[string]dispatcherHardening{
+	"xor":            xorHardening{},
+	"delegate_table": delegateTableHardening{},
+}
+
+func newDispatcherHardening(names []string) dispatcherHardening {
+	if len(names) == 0 {
+		return nil
```

Comment:
You panic for know names shouldn't this also panic then?

Reply:
is an optional parameter. Refactored to make it easier to read